### PR TITLE
John conroy/sort configure table

### PR DESCRIPTION
**Pull request by john-conroy**
Sorts the configure table by 'General' non-metadata fields then entity type primarily and alphabetically secondarily. 

@mccalluc I removed 'version' from the available fields, do you know of other fields that we might not want available? 

Screen Shot 2022-05-05 at 10 20 54 AM

### CODE REVIEW

File: context/app/static/js/components/entity-search/sidebar/ConfigureSearchTable/utils.js [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
-      return entityTypeA.localeCompare(entityTypeB) || fieldNameA.localeCompare(fieldNameB);
-    })
-    .map(([fieldName, fieldEntityType]) => ({ fieldName, fieldEntityType }));
+function getFieldEntriesSortedByConfigureGroup(fields) {
+  return Object.entries(fields).sort(
+    (
+      [, { configureGroup: configureGroupA, label: labelA }],
+      [, { configureGroup: configureGroupB, label: labelB }],
+    ) => {
+      // put 'General' configure group at top
```
Comment (mccalluc) [inline context/app/static/js/components/entity-search/sidebar/ConfigureSearchTable/utils.js:7]:
```suggestion
      // Sort 'General' configure group to top:
```
Capitalizing the first letter of comments, like I mentioned before. :)

File: context/app/static/js/components/entity-search/sidebar/DatasetConfigureSearchTable/hooks.js [hunk 2]
```diff
@@ -24,13 +24,17 @@ function selectedItemsReducer(state, { type, payload }) {
   }
 }
 
-function useGroupedMetadataFieldConfigs() {
+function useGroupedFieldConfigs() {
+  const { initialFields, initialFacets } = useStore();
   const metadataFieldConfigs = useMetadataFieldConfigs();
-  return Object.entries(metadataFieldConfigs).reduce((acc, [metadataFieldName, metadataFieldConfig]) => {
+
+  // Order matters. Field configs defined in initialFields/initialFacets will be overwritten
```
Comment (mccalluc) [inline context/app/static/js/components/entity-search/sidebar/DatasetConfigureSearchTable/hooks.js:31]:
```suggestion
  // Order matters. Field configs defined in initialFields/initialFacets will be overwritten.
```
and periods on sentences.

File: context/app/static/js/components/entity-search/sidebar/DatasetConfigureSearchTable/hooks.js [hunk 3]
```diff
@@ -58,13 +60,15 @@ function useFieldGroups() {
 }
 
 function buildFieldConfigs(fieldNames) {
-  return fieldNames.reduce(
-    (acc, fieldName) => ({
+  return fieldNames.reduce((acc, fieldName) => {
+    if (fieldName === 'version') {
```
Comment (mccalluc) [inline context/app/static/js/components/entity-search/sidebar/DatasetConfigureSearchTable/hooks.js:64]:
```suggestion
    const excludedFieldNames = ['version'];
    if (excludedFieldNames.includes(fieldName) {
```
In response to your question at the top: no others come to mind... but there probably will be more, so let's make it easier to find the code that does the work.

**Review by mccalluc — COMMENTED** [review]
Looks good! The fall-back logic in the sort is tidy. A couple suggestions, and it's good to merge.

File: context/app/static/js/components/entity-search/sidebar/ConfigureSearchTable/utils.js [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
-  return Object.entries(fields)
-    .sort(([fieldNameA, entityTypeA], [fieldNameB, entityTypeB]) => {
-      return entityTypeA.localeCompare(entityTypeB) || fieldNameA.localeCompare(fieldNameB);
-    })
-    .map(([fieldName, fieldEntityType]) => ({ fieldName, fieldEntityType }));
+function getFieldEntriesSortedByConfigureGroup(fields) {
+  return Object.entries(fields).sort(
+    (
+      [, { configureGroup: configureGroupA, label: labelA }],
+      [, { configureGroup: configureGroupB, label: labelB }],
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply context/app/static/js/components/entity-search/sidebar/ConfigureSearchTable/utils.js:5]: This is what ESlint prefers.

**Pull request merged**